### ISSUE TITLE
What you are claiming in the Blog is not seems to be true

### ISSUE THREAD

**Issue by akshat-suwalka** [microsoft/SynapseML#1868]
### SynapseML version

Version: 0.11.0

### System information
10.4 LTS ML (includes Apache Spark 3.2.1, Scala 2.12)
com.microsoft.azure:synapseml_2.12:0.10.1
pyspark in databricks

### Describe the problem

[https://microsoft.github.io/SynapseML/docs/features/responsible_ai/Interpretability%20-%20PDP%20and%20ICE%20explainer/#visualization](url)

This is the above page in which you are claiming that "The goal of this notebook is to show how these methods work for a pretrained model."

But in the following code you are fitting the model....which will train it....
`pipeline = Pipeline(
    stages=[
        StringIndexer()
        .setInputCol("income")
        .setOutputCol("label")
        .setStringOrderType("alphabetAsc"),
        StringIndexer()
        .setInputCols(categorical_features)
        .setOutputCols(string_indexer_outputs),
        OneHotEncoder()
        .setInputCols(string_indexer_outputs)
        .setOutputCols(one_hot_encoder_outputs),
        VectorAssembler(
            inputCols=one_hot_encoder_outputs + numeric_features, outputCol="features"
        ),
        GBTClassifier(weightCol="fnlwgt", maxDepth=7, maxIter=100),
    ]
)

model = pipeline.fit(df)`

How can we use the whole code without training it or fitting it....I have already pretrained model and i don't want it to further train....

### Code to reproduce issue

model_object = RandomForestClassificationModel.load(model_path)
pipeline  = Pipeline(stages = [VectorAssembler(inputCols=columns_input, outputCol="features"),model_object])

#if used then will throw error 'Pipeline' object has no attribute 'transform'
pipeline.transform(features_df) 

pdp = ICETransformer(
    model=pipeline,
    targetCol="probability",
    kind="average",
    targetClasses=[1],
    categoricalFeatures = ['football_played'],
    numericFeatures=['pd2_n_win'],
)

# will throw error - cannot pickle '_thread.RLock' object
output_pdp = pdp.transform(features_df)
display(output_pdp)

### What component(s) does this bug affect?

- [ ] `area/cognitive`: Cognitive project
- [X] `area/core`: Core project
- [ ] `area/deep-learning`: DeepLearning project
- [ ] `area/lightgbm`: Lightgbm project
- [ ] `area/opencv`: Opencv project
- [ ] `area/vw`: VW project
- [X] `area/website`: Website
- [ ] `area/build`: Project build system
- [X] `area/notebooks`: Samples under notebooks folder
- [ ] `area/docker`: Docker usage
- [X] `area/models`: models related issue

### What language(s) does this bug affect?

- [ ] `language/scala`: Scala source code
- [X] `language/python`: Pyspark APIs
- [ ] `language/r`: R APIs
- [ ] `language/csharp`: .NET APIs
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/synapse`: Azure Synapse integrations
- [ ] `integrations/azureml`: Azure ML integrations
- [X] `integrations/databricks`: Databricks integrations

**Comment by akshat-suwalka** [comment]
@svotaw

**Comment by memoryz** [comment]
@akshat-suwalka , to interpret a trained model, you can simply load the model with `PipelineModel.load(path)` (assuming it's a Spark model), and then pass the loaded model to the ICETransformer class. You don't have to train your model in the same notebook. The notebook included training steps so it's easier to demonstrate a typical scenario.

**Comment by akshat-suwalka** [comment]
Hi @memoryz thank you for the reply..
I tried that but it is showing the below error....Ideally it shouldn't be because it is spark model....

`requirement failed: Error loading metadata: Expected class name org.apache.spark.ml.PipelineModel but found class name org.apache.spark.ml.classification.RandomForestClassificationModel`

**Comment by memoryz** [comment]
> Hi @memoryz thank you for the reply.. I tried that but it is showing the below error....Ideally it shouldn't be because it is spark model....
> 
> When i tried to load the model... `model_object = PipelineModel.load(model_path)` below is the error `requirement failed: Error loading metadata: Expected class name org.apache.spark.ml.PipelineModel but found class name org.apache.spark.ml.classification.RandomForestClassificationModel`

Sorry for the confusion. What I meant in the previous reply is that you can directly pass the `model_object` (from `RandomForestClassificationModel.load(model_path)`) to the explainer:
```python
model_object = RandomForestClassificationModel.load(model_path)

pdp = ICETransformer(
    model=model_object,
    targetCol="probability",
    kind="average",
    targetClasses=[1],
    categoricalFeatures = ['football_played'],
    numericFeatures=['pd2_n_win'],
)
```
Your model is a `RandomForestClassificationModel`, not a `PipelineModel`, so if you try to load the model path as a `PipelineModel`, you would get exactly this error: `Expected class name org.apache.spark.ml.PipelineModel but found class name org.apache.spark.ml.classification.RandomForestClassificationModel`.

That said, you should not use the explainer in this way, and here is why:
The RandomForestClassificationModel expects feature vectors (which are commonly indexed/encoded by string indexer and onehot encoder), so the explainer can only perturb these vector features and treat them numerically. So the resulting interpretation will also be based on indexed/encoded features, which are not easy to comprehend. Furthermore, OneHot encoder outputs a binary vector where only 1 component is active at a time, but the explainer has no knowledge of that, and does not assume this rule, and it's going to randomly sample from the perturbation space and will get invalid samples (e.g. more than 1 component is simultaneously active).

The recommended way to explain your model is: In your training pipeline, package your featurizers (indexer, encoder, etc.) together with the classifier into a pipeline, fit the pipeline, and save the entire pipeline model. When you interpret the model, you load the entire pipeline model (`PipelineModel.load("path_to_pipeline_model")`), and explain it as a blackbox model.
